Interactive command line configuration (#53)
sapegin/mrm: docs/Writing_tasks.md

Create either `~/.mrm/<TASK>/index.js` or `~/dotfiles/mrm/<TASK>/index.js`. If `<TASK>` is the same as one of the default tasks your task will override a default one.

> ## Simple example

Reviewer: ```suggestion
## Basic example
```

[mrm-core](../packages/mrm-core) is an utility library created to write Mrm tasks, it has functions to work with common config files (JSON, YAML, INI, Markdown), npm dependencies, etc.

> ## With `mrm-core` and paramenters

Reviewer: ```suggestion
## With `mrm-core` and parameters
```

  const { eslintPreset } = config.values();

  // 4 - Define packages to install.
  const packages = ['eslint'];

  // 5 - Append custom eslint-config in case defined.
  if (eslintPreset !== 'eslint:recommended') {
    packages.push(`eslint-config-${eslintPreset}`);
  }

  // 6 - Create or load .eslintignore, and set basic ignores.
  lines('.eslintignore')
    .add(['node_modules/'])
    .save();

  // 7 - Create or load package.json.
  const pkg = packageJson();

  pkg
    // 8 - Set linting script.
    .setScript('lint', 'eslint . --cache --fix')
    // 9 - Set pretest script.
    .prependScript('pretest', 'npm run lint')
    // 10 - Save changes to package.json.
    .save();

  // 11 - Create or load .eslintrc.
  const eslintrc = json('.eslintrc');

  // 12 - Use Babel parser if the project depends on Babel.
  if (pkg.get('devDependencies.babel-core')) {
    const parser = 'babel-eslint';
    packages.push(parser);
    eslintrc.merge({ parser });
  }

  // 13 - Configure ESlint preset, if set (defaults to eslint:recommended).
  if (eslintPreset) {
    eslintrc.set('extends', eslintPreset);
  }

  // 14 - Save changes to .eslintrc.
  eslintrc.save();

>   // 15 - Install new NPM dependencies.

Reviewer: The official spelling is _npm_.

```suggestion
  // 15 - Install new npm dependencies.
```

  install(packages);
}

> // 16 - Configure task static information (used by Mrm CLI).

Reviewer: Maybe more clear:

```suggestion
// 16 - Configure task metadata (used by the `mrm` tool).
```

task.parameters = parameters;
task.description = 'Adds ESLint';

module.exports = task;
```

> Have a look at [`mrm-core` docs](../packages/mrm-core#api) for more utility helpers, and [the default tasks](../Readme.md#tasks) for reference.

> ### Configuration definition

Reviewer: They call them _prompts_ in the docs, and I think it's much clearer than _definitons_.

https://github.com/enquirer/enquirer

> The example above showcases a simple configuration prompt (`parameters`). More complex needs can be accomplished by defining prompts according to [enquirer docs](https://github.com/enquirer/enquirer). A simple required `name` prompt, for instance, could be defined as this:

Reviewer: ```suggestion
The example above shows a basic configuration prompt (`parameters`), but Mrm supports more complex prompts too. Have a look at [Enquirer docs](https://github.com/enquirer/enquirer).

For example, a required `name` prompt could be defined as this:
```
